packages/trader/src/Modules/Trading/Components/Form/DatePicker/trading-date-picker.tsx

```diff
@@ -126,13 +126,21 @@ const TradingDatePicker = observer(({ id, is_24_hours_contract, mode, name }: TT
         async (e = toMoment().format('YYYY-MM-DD')) => {
             const new_market_events: TMarketEvent[] = [];
             let new_disabled_days: number[] = [];
-            const events = await ContractType.getTradingEvents(e, symbol);
+
+            const [events, trading_days] = await Promise.all([
+                ContractType.getTradingEvents(e, symbol),
+                ContractType.getTradingDays(e, symbol),
+            ]);
+
+            if (trading_days) {
+                const all_days = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'] as const;
+                new_disabled_days = all_days
```

Comment:
I think we can do map and filtering together, maybe something like:
```
    new_disabled_days = all_days
        .reduce<number[]>((disabled, day, index) => {
            return trading_days.includes(day) ? disabled : [...disabled, index];
        }, []);
```
Please, ignore if irrelevant. It might decrease the code readability